src/gamemode.c

```diff
@@ -535,6 +535,47 @@ handle_unregister_game_by_pidfd (XdpDbusGameMode *object,
   return G_DBUS_METHOD_INVOCATION_HANDLED;
 }
 
+
+/* properties */
+static void
+update_active_state (GVariant *client_count)
+{
+  gboolean enabled = g_variant_get_int32 (client_count) > 0;
+  xdp_dbus_game_mode_set_active (XDP_DBUS_GAME_MODE (gamemode), enabled);
+}
+
+static void
+update_active_state_from_cache (GDBusProxy *proxy)
+{
+  g_autoptr(GVariant) client_count = NULL;
+
+  client_count = g_dbus_proxy_get_cached_property (proxy, "ClientCount");
+
+  if (client_count != NULL)
+    update_active_state (client_count);
```

Comment:
I guess you can update it in place here, instead of using a separate function.

Reply:
I'm going to push back a little on this, but if you still disagree, I'm happy to inline this method.

I like having this method be separate because it has the "business logic" of greater-than-zero. Which is admittedly trivial logic. But still, it felt appropriate to have only one place that knew how to map ClientCount to Active state.

But it's not important, and I'm happy to inline.